src/d3a/d3a_core/sim_results/cumulative_grid_trades.py

```diff
@@ -34,12 +58,297 @@ def __init__(self):
+            if area_sells_to_child(trade, area['name'], child_names):
+                accumulated_trades[area['name']]["consumedFromExternal"] = \
+                    subtract_or_create_key(accumulated_trades[area['name']]
+                                           ["consumedFromExternal"],
+                                           area_name_from_area_or_iaa_name(trade['buyer']),
+                                           trade['energy'])
+                accumulated_trades[area['name']]["spentToExternal"] = \
+                    add_or_create_key(accumulated_trades[area['name']]["spentToExternal"],
+                                      area_name_from_area_or_iaa_name(trade['buyer']),
+                                      trade_price)
+            elif child_buys_from_area(trade, area['name'], child_names):
+                accumulated_trades[area['name']]["producedForExternal"] = \
+                    add_or_create_key(accumulated_trades[area['name']]["producedForExternal"],
+                                      area_name_from_area_or_iaa_name(trade['seller']),
+                                      trade['energy'])
+                accumulated_trades[area['name']]["earnedFromExternal"] = \
+                    add_or_create_key(accumulated_trades[area['name']]["earnedFromExternal"],
+                                      area_name_from_area_or_iaa_name(trade['seller']),
+                                      trade_price)
+
+        accumulated_trades = CumulativeGridTrades._area_trade_from_parent(
+            area, parent, flattened_area_core_stats_dict, accumulated_trades
+        )
+
+        return accumulated_trades
+
+    @classmethod
+    def generate_cumulative_grid_trades_for_all_areas(cls, accumulated_trades, area,
+                                                      parent, results):
+        if area['children'] == []:
+            return results
+
+        results[area['uuid']] = [CumulativeGridTrades.generate_area_cumulative_trade_redis(
+                child, area, accumulated_trades
+            )
+            for child in area['children']
+            if child['name'] in accumulated_trades
+        ]
+        if parent is not None:
+            results[area['uuid']].append(CumulativeGridTrades._external_trade_entries(
+                area, accumulated_trades))
+
+        for child in area['children']:
+            results = cls.generate_cumulative_grid_trades_for_all_areas(
+                accumulated_trades, child, area, results
+            )
+        return results
+
+    @classmethod
+    def _area_trade_from_parent(cls, area, parent, flattened_area_core_stats_dict,
+                                accumulated_trades):
+        area_IAA_name = make_iaa_name_from_dict(area)
+        parent_trades = flattened_area_core_stats_dict.get(parent['uuid'], {}).get('trades', [])
+
+        for trade in parent_trades:
+            trade_price = trade['energy'] * trade['energy_rate']
```

Comment:
This is a strong point in favor of including the trade price in the trade JSON objects.

Reply:
Okay, will add now trade price in the trade JSON objects, the reason I was skeptical in first is to avoid sending redundant data via communication channels (in our case now it is redis) to save some data cache which could easily be calculated by result service using the already existing infos which is energy and rate here

Reply:
I agree with Faizan. Is it really worth to send and store a redundant information in the DB to avoid a multiplication operation?

Reply:
It depends how frequently do you want to show it. I agree that 'just' for calculating cumulative grid trades it is not worth it. However, if we ever need to display it in a graph (for instance in the bid/offer/trade graph) or if we want to feed the trade price in more stats than that then it is definitely worth to precalculate (assuming that DB space is not the issue there) instead of having to have a separate step that would perform this multiplication. 
Having said that, since we only have this ATM I agree to not add the price yet. However I predict such a need will appear soon

Reply:
ok no, if you put it like this, we can add the price indeed.

Reply:
So, I will keep the pre-calculated `trade_price ` as part of trade JSON object and will see later if it makes sense to be pre-calculated or some DB space needs to be saved and calculated on the fly.